tests/utils/test_endpoints.py

```diff
@@ -94,3 +94,19 @@ def test_endpoint_config_custom_token_name():
     actual = endpoint_utils.EndpointConfig.from_dict(test_data)
 
     assert actual.token_name == "test_token"
+
+
+async def test_request_non_json_response():
+    with aioresponses() as mocked:
+        endpoint = endpoint_utils.EndpointConfig("https://example.com/")
+
+        mocked.post(
+            "https://example.com/test",
+            payload="ok",
+            content_type="application/text",
+            status=200,
+        )
+
+        response = await endpoint.request("post", subpath="test",)
+
+        assert response
```

Comment:
I guess that should be `assert not response`.
We're returning None in case the response doesn't have json.

Reply:
Yeah, realized this after I had changed the implementation 😃wasn't super concerned about the tests before i got feedback on what you thought about it as a patch.